src/waitFor.js

```diff
@@ -38,6 +49,25 @@ export default function waitFor<T>(
   });
 }
 
+export default async function waitFor<T>(
+  expectation: () => T,
+  options?: WaitForOptions
+): Promise<T> {
+  if (!checkReactVersionAtLeast(16, 9)) {
+    return await waitForInternal(expectation, options);
```

Comment:
`return await` is redundant, you can omit the `await`

Reply:
Removed await